### DISCUSSION TITLE
Type mismatch in some level-related analytics API methods?

### DISCUSSION THREAD

**Discussion by chrisbartley** [invertase/react-native-firebase#6040] [Q&A]
A couple of the API methods for level-related analytics--e.g. [`logLevelStart`](https://rnfirebase.io/reference/analytics#logLevelStart) and [`logLevelEnd`](https://rnfirebase.io/reference/analytics#logLevelEnd)--specify level should be a number.  However, GA4's documentation for the[`level_start`](https://developers.google.com/analytics/devguides/collection/ga4/reference/events#level_start) and [`level_end`](https://developers.google.com/analytics/devguides/collection/ga4/reference/events#level_end) say that the `level_name` parameter is a string.

It seems that GA4's API is a little inconsistent here, because, for example, the [`level_up`](https://developers.google.com/analytics/devguides/collection/ga4/reference/events#level_up) and [`post_score`](https://developers.google.com/analytics/devguides/collection/ga4/reference/events#post_score) events both expect a numeric `level` parameter.

So what do people actually do here, in practice?  Are these types actually enforced on either or both ends?  Thanks!

**Comment by mikehardy** [comment]
Hi there - hopefully someone with more direct experience will chime in (please :pray: ) - but since I'm the active maintainer just want to say that I've not logged these personally. All I can suggest in the absence of other people chiming in is direct experimentation to see if other types go through successfully or are coerced or similar

**Reply by chrisbartley** [reply]
Thanks for the quick followup.  I'm not sure about enforcement on the GA4 side of things, but a quick test of RN Firebase's `logLevelStart` method confirms that giving a non-numeric value for the `level` parameter does indeed throw an error, e.g.:

```
TypeError: firebase.analytics().logLevelStart(*): Expected a value of type `number` for `level` but received `"Level one"`.
```

Numeric values post through just fine, presumably--if GA4's docs are correct about wanting a string--because there's some string conversion going on behind the scenes.

Anyway, thanks for the answer, and my apologies for not just trying it out to begin with.  I like that RN Firebase's API is at least consistent with regards to `level` always being a number, but in practice it feels a bit limiting.  But I also totally understand that it may simply be the most logical response to GA4's "sometimes it's a number, sometimes it's a string" API 🤷‍♂️   Thanks!
